tests/unit/testServiceConfiguration.js

```diff
@@ -24,546 +24,546 @@ var logger = require('../../src/logger.js')({
   logLevel: isNumber(level) ? level : 4
 });
 
-function makeTest(serviceName, serviceVersion,
-                  moduleName, moduleVersion,
-                  functionName, testFunction) {
-  return function(t) {
-    // store the old environment
-    var oldServiceName = process.env.GAE_SERVICE || '';
-    var oldServiceVersion = process.env.GAE_VERSION || '';
+describe('Testing service configuration', function () {
+  function makeTest(serviceName, serviceVersion,
```

Comment:
Instead of dynamically generating the functions passed to `it`, I would suggest putting `it` test cases in a for loop iterating over all of the combinations of arguments being tested: http://mochajs.org/#dynamically-generating-tests